File: 1588-sum-of-all-odd-length-subarrays/1588-sum-of-all-odd-length-subarrays.py
class Solution:
    def sumOddLengthSubarrays(self, arr: List[int]) -> int:
        n = len(arr)
        ans = 0
        for i in range(n):
            curr_total = 0
            for j in range(i, n):
                curr_total += arr[j]
                if (j - i + 1) & 1:
                    ans += curr_total
        return ans

File: 1588-sum-of-all-odd-length-subarrays/1588-sum-of-all-odd-length-subarrays.java
class Solution {
    public int sumOddLengthSubarrays(int[] arr) {
        int ans = 0;
        for(int i = 0; i < arr.length; i++){
            int total = 0;
            for(int j = i; j < arr.length; j++){
                total += arr[j];
                if((i - j) % 2 == 0)
                    ans += total;
            }
        }
        return ans;
    }
}